casepro/msgs/views.py

```diff
@@ -5,27 +5,47 @@
 from collections import defaultdict
 from dash.orgs.views import OrgPermsMixin, OrgObjPermsMixin
 from django.http import HttpResponse, HttpResponseBadRequest, JsonResponse
+from django.core.urlresolvers import reverse
 from django.utils.timesince import timesince
 from django.utils.translation import ugettext_lazy as _
+from django.utils.timezone import now
 from el_pagination.paginators import LazyPaginator
 from smartmin.mixins import NonAtomicMixin
 from smartmin.views import SmartCRUDL, SmartTemplateView
-from smartmin.views import SmartListView, SmartCreateView, SmartReadView, SmartUpdateView, SmartDeleteView
+from smartmin.views import (SmartListView, SmartCreateView, SmartUpdateView, SmartDeleteView, SmartReadView,
+                            SmartCSVImportView)
+from smartmin.csv_imports.models import ImportTask
 from temba_client.utils import parse_iso8601
 
 from casepro.rules.mixins import RuleFormMixin
 from casepro.statistics.models import DailyCount
-from casepro.utils import parse_csv, str_to_bool, JSONEncoder, json_encode, month_range
+from casepro.utils import parse_csv, str_to_bool, JSONEncoder, json_encode, month_range, get_language_name
 from casepro.utils.export import BaseDownloadView
 
-from .forms import LabelForm
-from .models import Label, Message, MessageExport, MessageFolder, Outgoing, OutgoingFolder, ReplyExport
+
+from .forms import LabelForm, FaqForm
+from .models import Label, FAQ, Message, MessageExport, MessageFolder, Outgoing, OutgoingFolder, ReplyExport
 from .tasks import message_export, reply_export
 
 
 RESPONSE_DELAY_WARN_SECONDS = 24 * 60 * 60  # show response delays > 1 day as warning
 
 
+# Override the ImportTask start method so we can use our self-defined task
+def override_start(self, org):
+    from .tasks import faq_csv_import
+    self.log("Queued import at %s" % now())
+    self.save(update_fields=['import_log'])
+
+    # trigger task
+    result = faq_csv_import.delay(org, self.pk)
+
+    self.task_id = result.task_id
+    self.save(update_fields=['task_id'])
+
+ImportTask.start = override_start
```

Comment:
That is all WCL's code. I just fixed the tests and refactored the language model into a field